discovery-provider/src/models/models.py

```diff
@@ -261,10 +261,25 @@ class Track(Base):
         viewonly=True,
     )
 
+    _owner = relationship(
+        "User",
+        primaryjoin="and_(\
+            remote(Track.owner_id) == foreign(User.user_id),\
+            User.is_current)",
+        lazy="joined",
+        viewonly=True,
+    )
+
     @property
-    def slug(self):
+    def _slug(self):
         return self._routes[0].slug if self._routes else ""
 
+    @property
+    def permalink(self):
+        if self._owner and self._owner[0].handle and self._slug:
+            return f"/{self._owner[0].handle}/{self._slug}"
```

Comment:
any issue w/ there being /handle/"" if for some reason there is a slug but no users? i dont think possible, but figured i'd call out

Reply:
if `_slug` is an empty string it's falsy, right? Goal would to be if either are empty/null, the permalink is empty string.

Reply:
yes yes you're right